modules/pam_shells/pam_shells.c

```diff
@@ -13,27 +13,48 @@
 #include <string.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <stdbool.h>
 #include <sys/stat.h>
 #include <syslog.h>
 #include <unistd.h>
+#if defined (USE_ECONF)	&& defined (VENDORDIR)
+#include <libeconf.h>
+#endif
 
 #include <security/pam_modules.h>
 #include <security/pam_modutil.h>
 #include <security/pam_ext.h>
 
 #define SHELL_FILE "/etc/shells"
-
+#define SHELLS "shells"
+#define ETCDIR "/etc"
 #define DEFAULT_SHELL "/bin/sh"
 
+static bool check_file(const char *filename, const void *data)
+{
+    struct stat sb;
+    const pam_handle_t *log_pamh = (const pam_handle_t *)data;
```

Comment:
I don't think you need an explicit cast from `void *` to `pam_handle_t *`.

Reply:
Yes, you are right. I am wondering why the compiler does not complain it.